ext/AdvancedHMCComponentArraysExt.jl

```diff
@@ -0,0 +1,39 @@
+module AdvancedHMCComponentArraysExt
+
+using LinearAlgebra
+
+using AdvancedHMC:
+    AdvancedHMC,
+    Hamiltonian,
+    UnitEuclideanMetric,
+    DiagEuclideanMetric,
+    DenseEuclideanMetric,
+    GaussianKinetic
+using ComponentArrays: ComponentVecOrMat, ComponentVector, ComponentMatrix, getaxes
+
+function AdvancedHMC.∂H∂r(
```

Comment:
We don't need to overload these functions `∂H∂r`. It seems all we need is to define a utility function `AHMC.__axes`, then overload `AHMC.__axes(x::ComponentVecOrMat) = getaxes(x)` in this `AHMCComponentArrays` extension.